### PR TITLE
add i18next-xhr-backend definitions

### PR DESCRIPTION
**Pull request by bryanjjohnson**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url https://github.com/i18next/i18next-xhr-backend .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

**Comment by andy-ms** [comment]
* The url from your comment (https://github.com/i18next/i18next-xhr-backend) would make a better `Project` link.
* The type definition takes any object as options, but has a property `options` of a more specific type. The `options` in your test aren't really being type-checked.

Pinging @mxl and @cyrilschumacher who might know how i18next typings work.

File: i18next-xhr-backend/i18next-xhr-backend.d.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        options: {
+            loadPath: string | Function,
+            addPath:  string,
+            allowMultiLoading: boolean,
+            parse: () => {},
+            crossDomain: boolean,
+            withCredentials: boolean,
+            ajax: () => void
+        };
+        constructor(services: Services, options?: {});
```
Comment (cyrilschumacher) [inline i18next-xhr-backend/i18next-xhr-backend.d.ts]: `services` can be optional.
See example from [README.md](https://github.com/i18next/i18next-xhr-backend#backend-options).

File: i18next-xhr-backend/i18next-xhr-backend.d.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+declare module 'i18next-xhr-backend' {
+
+    interface Interpolator {
+        interpolate: () => string
+    }
+    interface Services {
+        interpolator: Interpolator
+    }
+
+    export default class Backend {
```
Comment (cyrilschumacher) [inline i18next-xhr-backend/i18next-xhr-backend.d.ts]: At first sight, some properties are optional.
See [index.js line 4](https://github.com/i18next/i18next-xhr-backend/blob/master/src/index.js#L4).

**Review by cyrilschumacher — COMMENTED** [review]
@bryanjjohnson I did a quick review to improve your definition, otherwise good work :)

**Comment by bryanjjohnson** [comment]
Got some fixes in

**Comment by mxl** [comment]
@bryanjjohnson  @andy-ms Looks good.

**Pull request merged**